code/Forms/UploadField.php

```diff
@@ -326,28 +328,52 @@ public function performDisabledTransformation()
 
     /**
      * Checks if the number of files attached adheres to the $allowedMaxFileNumber defined
+     * Also checks that extensions are valid, which is intended primarily for selecting
+     * existing files rather than newly uploaded files, as those will be checked by
+     * Upload_Validator which is used on the UploadReciever trait
      *
      * @param Validator $validator
      * @return bool
      */
     public function validate($validator)
     {
         $maxFiles = $this->getAllowedMaxFileNumber();
-        $count = count($this->getItems() ?? []);
-
-        if ($maxFiles < 1 || $count <= $maxFiles) {
-            return $this->extendValidationResult(true, $validator);
+        /** @var SS_List<File> $items */
+        $items = $this->getItems();
+        $count = count($items ?? []);
```

Comment:
```suggestion
        $count = count($items);
```
A new `ArrayList` is returned if there's no items, so this won't ever be null

Reply:
Canged to $items->count() since it's an SS_List